common/src/sync.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Valued\PrestaShop;
+
+use Configuration;
+use DateTime;
+use Exception;
+use ModuleFrontController;
+use PrestaShop\Module\ProductComment\Entity\ProductComment;
+use PrestaShop\PrestaShop\Adapter\Validate;
+use Product;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class sync extends ModuleFrontController {
+
+    /** @var bool */
+    public $ajax;
+
+    /**
+     * @throws Exception
+     */
+    public function postProcess() {
+        $request_data = trim(file_get_contents('php://input'));
+        if (!$request_data) {
+            throw new HttpException(400, 'Empty request data');
+        }
+        if (!$request_data = json_decode($request_data, true)) {
+            throw new HttpException(400, 'Invalid JSON data provided');
+        }
+
+        if (
+            !$this->hasCredentialFields($request_data['webshop_id'], $request_data['api_key'])
+            || $this->credentialsEmpty($request_data['webshop_id'], $request_data['api_key'])
+        ) {
+            throw new HttpException(403, 'Missing credential fields');
+        }
+        $lang_id = (int) Configuration::get('PS_LANG_DEFAULT');
+        $product = new Product($request_data['product_review']['product_id'], false, $lang_id);
+        if (!Validate::isLoadedObject($product)) {
+            throw new HttpException(404, sprintf('Could not find product with ID (%d)', $request_data['product_review']['product_id']));
+        }
+
+        if (!Configuration::get(strtoupper($this->module->name) . '_SYNC_PROD_REVIEWS')) {
+            throw new HttpException(403, 'Product review sync is disabled.');
+        }
+
+        $this->isAuthorized($request_data['webshop_id'], $request_data['api_key']);
```

Comment:
I would move this check a bit higher, after `hasCredentialFields` and `credentialsEmpty`